docs/cody/capabilities/ignore-context.mdx

```diff
@@ -1,22 +1,157 @@
+</video>
+
+The `include` rules allow Cody to access the repository matching the `regexp`. This means Cody can now fully access the repository and its content to fetch context.
+
+3. Only `exclude` rules are specified
+
+All repositories are considered included by default without the `include` rules. Cody is prohibited from using content from any repository whose name matches at least one pattern specified in the exclude field. An `exclude` field may contain a catch-all condition (e.g., a `regexp` matching any string). If such a match occurs, Cody is restricted from using content from any repository. For example,
+
+```
+{
+ 'cody.contextFilters': {
+ // All repositories are allowed except for "github.com/sourcegraph/cody-analytics"
+ // or the ones containing "secret" in their name.
+ exclude: [
+ {repoNamePattern: '^github\\.com\\/sourcegraph\\/cody-analytics$'},
+ {repoNamePattern: '.*secret.*'}
+ ]
+ }
+}
+```
+
+![commands-disabled-exclude-rules](https://storage.googleapis.com/sourcegraph-assets/Docs/commands-disabled.png)
+
+In this case, Cody's commands are disabled, and you cannot use them for context fetching. If you try running any of these, you'll be prompted with an error message. However, Cody chat will still work, and you can use it to ask questions.
+
+<video width="1920" height="1080" loop playsInline controls style={{ width: '100%', height: 'auto' }}>
+  <source src="https://storage.googleapis.com/sourcegraph-assets/Docs/Media/exclude-rules.mp4" type="video/mp4"/>
+</video>
+
+4. Both `include` and `exclude` rules are specified
+
+When making requests to third-party LLMs, Cody can use content from a repository if its name matches any of the `include` patterns and does not match any of the `exclude` patterns. Thus, the `exclude` rules filter only the repositories allowed by the `include` rules. For example,
+
+```
+{
+ 'cody.contextFilters': {
+ // All repositories starting with "github.com/sourcegraph/" are allowed
+ include: [{repoNamePattern: '^github\\.com\\/sourcegraph\\/.+'}],
+ // except for "github.com/sourcegraph/cody-analytics"
+ // and the ones containing "cody" in their name.
+ exclude: [
+ {repoNamePattern: '^github\\.com\\/sourcegraph\\/cody-analytics$'},
+ {repoNamePattern: '.*cody.*'}
+ ]
+ }
+}
+```
+
+In this case, Cody can access content from repositories whose names start with `github.com/sourcegraph/`. However, it excludes files from repositories with `cody` in their name.
+
+## Context Filters Compatibility Matrix
+
+<Callout type="info"> Cody Context Filters work only with Sourcegraph versions >=5.4.0 for Enterprise users.</Callout>
+
+Depending on the client type, here's a detailed breakdown of all the prerequisites that are supported for the context filters to work.
+
+|        **Client Versions**        | Sourcegraph `v<5.4.0` | Sourcegraph `v=5.4.0`                           | Sourcegraph `v=5.4.X`                           | Sourcegraph `v>=5.4.0` |
```

Comment:
@kalanchan Do we have a release version for the SG instance with the prompt to upgrade notification?

Reply:
yes! the prompt to upgrade notification is in Sourcegraph versions >= 5.4.2198